cdap-app-fabric/src/main/java/io/cdap/cdap/internal/credential/DefaultCredentialProviderService.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.credential;
+
+import com.google.common.util.concurrent.AbstractIdleService;
+import io.cdap.cdap.api.security.credential.CredentialIdentity;
+import io.cdap.cdap.api.security.credential.CredentialProvisioningException;
+import io.cdap.cdap.api.security.credential.IdentityValidationException;
+import io.cdap.cdap.api.security.credential.NotFoundException;
+import io.cdap.cdap.api.security.credential.ProvisionedCredential;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.proto.credential.CredentialProfile;
+import io.cdap.cdap.proto.id.CredentialIdentityId;
+import io.cdap.cdap.proto.id.CredentialProfileId;
+import io.cdap.cdap.proto.security.StandardPermission;
+import io.cdap.cdap.security.spi.authorization.ContextAccessEnforcer;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+import javax.inject.Inject;
+
+/**
+ * Default implementation for {@link CredentialProviderService} used in AppFabric.
+ */
+public class DefaultCredentialProviderService extends AbstractIdleService
+    implements CredentialProviderService {
+
+  private final CConfiguration cConf;
+  private final ContextAccessEnforcer contextAccessEnforcer;
+  private final Map<String, io.cdap.cdap.security.spi.credential.CredentialProvider> credentialProviders;
```

Comment:
why not import io.cdap.cdap.security.spi.credential.CredentialProvider?

Reply:
Fixed, thanks!